### PR TITLE
Cosserat

### PR DESCRIPTION
**Pull request by alessiofumagalli**

### CODE REVIEW

**Comment by alessiofumagalli** [comment]
the pr is still a draft, we need to include some tests for the new parts and a tutorial

**Comment by alessiofumagalli** [comment]
@wmboon I've completed the test that are needed for this pr. let me know what you think

**Comment by alessiofumagalli** [comment]
also the tutorial has been updated, please have a look

File: src/pygeon/grids/create_grid.py [hunk 1]
```diff
@@ -25,12 +25,14 @@ def grid_from_domain(
     Returns:
         Either a pg.MixedDimensionalGrid or a pg.Grid, depending on the value of as_mdg.
     """
-    mesh_size_min = kwargs.get("mesh_size_min", mesh_size / 10)
+    as_mdg = kwargs.pop("as_mdg", True)
```
Comment (wmboon) [inline src/pygeon/grids/create_grid.py:28]: Why is the keyword popped here? Should it not be kwargs.get() ?

File: tests/unit/test_vecpwconstant.py [hunk 2]
```diff
@@ -179,8 +179,13 @@ def test_error_l2(self):
...
         ana_sol = lambda x: np.sin(x)
 
+        err = discr.error_l2(sd, num_sol, ana_sol, etype="standard")
+        err_known = 7.951651025069553e-08
+
+        self.assertTrue(np.allclose(err, err_known))
+
         err = discr.error_l2(sd, num_sol, ana_sol)
-        err_known = 5.622666361455642e-08
+        err_known = 0.5254595621486393
```
Comment (wmboon) [inline tests/unit/test_vecpwconstant.py:188]: Let's double check why this error is so large

Comment (alessiofumagalli) [reply tests/unit/test_vecpwconstant.py:188]: cause the grid is very coarse and the function we are evaluating is the sin, when refining the error decays as expected. for this l2-error the computation of the analytical solution is done on the vertices and compared with the one in the cell centers. while for the other evaluation of the error (etype="standard") the analytical solution is interpolated over the cell centers

**Review by wmboon — APPROVED** [review]

**Pull request merged**